src/ucp/core/ucp_worker.c

```diff
@@ -2021,27 +2064,24 @@ void ucp_worker_destroy(ucp_worker_h worker)
     UCS_ASYNC_BLOCK(&worker->async);
     ucp_worker_destroy_eps(worker);
     ucp_worker_remove_am_handlers(worker);
-    ucp_am_cleanup(worker);
-    ucp_worker_close_cms(worker);
     UCS_ASYNC_UNBLOCK(&worker->async);
 
-    ucp_worker_destroy_ep_configs(worker);
+    ucp_am_cleanup(worker);
     ucp_tag_match_cleanup(&worker->tm);
-    ucs_mpool_cleanup(&worker->am_mp, 1);
-    ucs_mpool_cleanup(&worker->reg_mp, 1);
-    ucs_mpool_cleanup(&worker->rndv_frag_mp, 1);
+    ucp_worker_destroy_mpools(worker);
+    ucp_worker_destroy_mem_type_endpoints(worker);
+    ucp_worker_close_cms(worker);
     ucp_worker_close_ifaces(worker);
     ucp_worker_wakeup_cleanup(worker);
-    ucs_mpool_cleanup(&worker->rkey_mp, 1);
-    ucs_mpool_cleanup(&worker->req_mp, 1);
     uct_worker_destroy(worker->uct);
     ucs_async_context_cleanup(&worker->async);
-    ucs_conn_match_cleanup(&worker->conn_match_ctx);
-    kh_destroy_inplace(ucp_worker_rkey_config, &worker->rkey_config_hash);
-    ucs_ptr_map_destroy(&worker->ptr_map);
-    ucs_strided_alloc_cleanup(&worker->ep_alloc);
     UCS_STATS_NODE_FREE(worker->tm_offload_stats);
     UCS_STATS_NODE_FREE(worker->stats);
+    ucs_ptr_map_destroy(&worker->ptr_map);
+    ucs_strided_alloc_cleanup(&worker->ep_alloc);
+    kh_destroy_inplace(ucp_worker_rkey_config, &worker->rkey_config_hash);
+    ucp_worker_destroy_ep_configs(worker);
+    ucs_conn_match_cleanup(&worker->conn_match_ctx);
```

Comment:
logically `ucs_conn_match_cleanup` should be done before EPs and ptr_map clean up